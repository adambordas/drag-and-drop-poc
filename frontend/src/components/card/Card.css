.card {
  padding: 20px 30px;
  border: 1px solid #CCC;
}
