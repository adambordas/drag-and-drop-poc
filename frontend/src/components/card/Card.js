import './Card.css';

function Card() {
  const applyDragData = e => {
    e.dataTransfer.setData('card', e.target);
  };

  return (
    <div className="card" draggable onDragStart={applyDragData}>
      Todo item 1
    </div>
  );
}

export default Card;
