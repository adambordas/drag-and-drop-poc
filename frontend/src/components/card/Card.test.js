import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Card from './Card';

describe('Card component', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders properly', () => {
    const id = 'test id';
    const description = 'test description';

    const { container, getByText } = render(
      <Card id={id} description={description} />
    );

    const card = getByText(description);

    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('draggable');
    expect(container).toMatchSnapshot();
  });

  it('sets card id to dataTransfer object on dragStart event', () => {
    const mockSetData = jest.fn();
    const dataTransfer = {
      setData: mockSetData
    };

    const id = 'test id';
    const description = 'test description';

    const { getByText } = render(
      <Card id={id} description={description} />
    );

    const card = getByText(description);

    fireEvent.dragStart(card, { dataTransfer });

    expect(mockSetData).toHaveBeenCalledTimes(1);
    expect(mockSetData).toHaveBeenCalledWith('cardId', id);
  });
});
