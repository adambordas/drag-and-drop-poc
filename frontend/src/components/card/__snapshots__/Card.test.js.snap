// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders properly 1`] = `
<div>
  <div
    class="card"
    draggable="true"
  >
    test description
  </div>
</div>
`;
